src/ml_board/frontend/Dashboard/src/app/App.tsx

```diff
@@ -99,80 +101,82 @@ export default function App() {
     });
 
     useEffect(() => {
-        if(isConfigValidated)
-        {
+        if (isConfigValidated) {
             // save to local storage only after user clicks on submit button - either in popup or in settings page.
             // after the used submits the values & after it is saved, then only we will connect to the socket with the values given by user.
             // TODO:: after Vijul and Osama's code is merged, handle error message from the socket and show to user - so if the socket connection was not successful, user can update the values and try again.
             localStorage.setItem('SettingConfigs', JSON.stringify(settingConfigs));
 
-            // TODO: is DedicatedWorker really needed? 
-            const mlgymWorker = new DedicatedWorker(workerOnMessageHandler);
-            // NOTE: this is better than calling "useAppSelector(selectEvalResult)" as it will force the App function to get called everytime the state changes
-            // TODO: maybe find a better way later other than starting the worker with the empty redux state?
-            const evalResult = {
-                grid_search_id: null,
-                experiments: {},
-                colors_mapped_to_exp_id: [[], []]
+            // creating WebWorker
+            // NOTE:using URL because create-react-app throws error since it has not found the worker file during load/bundling
+            const workerSocket = new Worker(new URL('../worker_socket/WorkerSocket.ts', import.meta.url));
+            // setting the redux update methods on the incoming data from the worker thread
+            workerSocket.onmessage = ({ data }: MessageEvent) => workerOnMessageHandler(data as DataToRedux);
+            // starting the worker
+            workerSocket.postMessage(settingConfigs);
+
+            // // NOTE: this is better than calling "useAppSelector(selectEvalResult)" as it will force the App function to get called everytime the state changes
+            // // TODO: maybe find a better way later other than starting the worker with the empty redux state?
+            // const evalResult = {
+            //     grid_search_id: null,
+            //     experiments: {},
+            //     colors_mapped_to_exp_id: [[], []]
+            // }
+            // workerSocket.postMessage(evalResult);
+
+            // close the worker on Dismount to stop any memory leaks
+            return () => {
+                // ASK: not sure if this is useful, or if it gets handled before the termination???
```

Comment:
@vijulshah what do you think about this termination part here? Is this best practice?

Reply:
@le1nux  Yes this is the best practice to terminate a web worker.  On unmounting a component, it is done synchronously inside `ComponentWillUnmount` lifecycle hook or inside `useEffect` hook (otherwise, there will be a zombie web worker instances alive). 
Although, in our case it doesn't matter much because our component will only unmount on closing the app (as it is the main file i.e `App.tsx`). But still it's good to have, no harm in going according to the best practices